bittensor/_synapse/__init__.py

```diff
@@ -101,6 +102,38 @@ def TextCausalLM (
             backward_response_serializer_type = backward_response_serializer_type,
         )
 
+    @staticmethod
+    def TextCausalLMNext(
+        topk: int = 4096,
+        forward_request_serializer_type: 'bittensor.proto.Serializer.Type' = bittensor.proto.Serializer.MSGPACK,
+        forward_response_serializer_type: 'bittensor.proto.Serializer.Type' = bittensor.proto.Serializer.MSGPACK,
+        backward_request_serializer_type: 'bittensor.proto.Serializer.Type' = bittensor.proto.Serializer.MSGPACK,
+        backward_response_serializer_type: 'bittensor.proto.Serializer.Type' = bittensor.proto.Serializer.MSGPACK,
+    ) -> TextCausalLMNext:
+        """ Factory function which returns a TextCausalLMNext synapse adapter given arguments.
+            Args:
+                topk (:obj:`int`):
+                    Specifies the number of topk server token phrases to return.
```

Comment:
A little bit confused about the topk variable. Are they the topk logits of the server's tokenizer?

Reply:
Yes, topk logits/phrases, under the assumption that each logit represents a text phrase (the server represents this with a single token with associated probability, but it could be multiple std tokens (a token phrase) at the validator).